src/main/java/org/junitpioneer/jupiter/params/IntegralToBytesConverter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.params;
+
+import static java.lang.String.format;
+
+import java.nio.ByteBuffer;
+
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.params.converter.ArgumentConversionException;
+import org.junit.jupiter.params.converter.ArgumentConverter;
+import org.junit.platform.commons.support.AnnotationSupport;
+
+class IntegralToBytesConverter implements ArgumentConverter {
+
+	public IntegralToBytesConverter() {
+		// recreate default constructor to prevent compiler warning
+	}
+
+	@Override
+	public byte[] convert(Object source, ParameterContext context) throws ArgumentConversionException {
+		java.nio.ByteOrder byteOrder = AnnotationSupport
+				.findAnnotation(context.getParameter(), IntegralToBytesConversion.class)
+				.map(IntegralToBytesConversion::byteOrder)
+				.map(ByteOrder::getValue)
+				.orElseThrow(() -> new IllegalStateException(
+					format("Could not find %s annotation on the parameter", IntegralToBytesConversion.class)));
```

Comment:
In my local attempt, I used [`AnnotationConsumer`](https://junit.org/junit5/docs/5.9.2/api/org.junit.jupiter.params/org/junit/jupiter/params/support/AnnotationConsumer.html) to capture the target annotation in a private field (basically taking inspiration from [`JavaTimeArgumentConverter`](https://github.com/junit-team/junit5/blob/bd7d03f517dfca3730fbd123c5fd4d4b70930129/junit-jupiter-params/src/main/java/org/junit/jupiter/params/converter/JavaTimeArgumentConverter.java#L35-L36) of JUnit).

Something like the following:

```java

public class ByteArrayArgumentConverter extends TypedArgumentConverter<Number, byte[]>
		implements AnnotationConsumer<ByteArrayConversion> {

	private java.nio.ByteOrder order;

	ByteArrayArgumentConverter() {
		super(Number.class, byte[].class);
	}

	@Override
	public void accept(ByteArrayConversion byteArrayConversion) {
		switch (byteArrayConversion.order()) {
			case BIG_ENDIAN:
				order = java.nio.ByteOrder.BIG_ENDIAN;
				break;
			case LITTLE_ENDIAN:
				order = java.nio.ByteOrder.LITTLE_ENDIAN;
				break;
			default:
				throw new IllegalArgumentException("Unsupported byte order: " + byteArrayConversion.order());
		}
	}

	@Override
	protected byte[] convert(Number source) throws ArgumentConversionException {
		...
	}

```

(`java.nio.ByteOrder` can actually be imported as it doesn't clash with the `ByteOrder` from this package)